### DISCUSSION TITLE
Libre 3: should history fill in after my phone is disconnected?

### DISCUSSION THREAD

**Discussion by jmandel** [j-kaltes/Juggluco#137] [Q&A] [answered]
After my first night using Juggluco and a Libre 3, it seems like my phone disconnected or perhaps restarted overnight. I expect this should result in missing "stream" values -- but I thought that the sensor's stored memory of 5-minute history values would be retrievable and would still appear in the juggluco history plot.

(I have tried terminating the connection and reconnecting, toggling Bluetooth, rescanning NFC, etc.)

I'm including a screenshot in case there is some relevant information that I am not describing adequately! Happy to provide additional information if it is helpful.

![Screenshot_20240317-085959~2](https://github.com/j-kaltes/Juggluco/assets/313089/855f7642-ba1f-42e9-8f4d-d20ea342889c)

```
Version Code: 455
Version Name: 7.6.1
arm64-v8a
Build time: 14:41:57 07-Mar-2024
```

**Comment by j-kaltes** [comment]
Probably you had a sensor error. It there under Left menu->Sensor after "Glucose Value" a time and date under "Last failure"?

**Comment by jmandel** [comment]
There is currently a time listed from after I woke up, with a ~~strikethrough~~ in the formatting.

In the event of a failure, is there a way to interrogate the sensor to extract its historical log data? Or is this information gone by now?

**Comment by j-kaltes** [comment, marked as answer]
You can connect the sensor with Abbott's Libre 3 app and after waiting for the history  to be received by the app and send to Libreview, you can save the data to a file in Libreview and see what is present there.

**Comment by jmandel** [comment]
Thanks! I'm not sure how long to wait, but the graph in Libre 3 all (and download from Libreview site) have the same gap I see in Juggluco. I did have Juggluco uploading data to Libreview though, so maybe that's part of the issue.

Tried deleting the online Libreview data, uninstalling Libre 3 app, reinstalling, and reconnecting the sensor. The five hour gap persists, so I guess that suggests that the device really is missing the history?

**Reply by j-kaltes** [reply]
It is always best not to send data from Juggluco to the same Libreview account as those from Abbott's Libre 3 app. To do this you need to use Abbott's app Libreview account to retrieve the Account ID in Juggluco and thereafter save it as Manual under "Get Account ID" and fill in another Libreview account.
You can delete the data send by Juggluco by deleting its device from Account Settings->My devices in Libreview.

**Reply by jmandel** [reply]
Interesting, thanks! So one account ID is used for establishing a handshake with  the sensor (which needs to match the one that was used to initialize the sensor) but then Juggluco can upload to a different account in the libreview cloud system.

In any case I don't think I have any long-term use for Libreview or for the Libre Android app; I just went down that path to see if (for whatever reason) they could access the missing data from that overnight time window. But it seems like the answer is no.
